### ISSUE TITLE
Webcam causes ingame FPS Drop

### ISSUE THREAD

**Issue by ZePhyr1987** [UltraStar-Deluxe/USDX#279]
As soon as the FPS of the camera gets lower (lowlight), goes also the ingame FPS.
But with preset low FPS it is not quite as bad. (Resolution/Effect doesn't matter)
![screenshot0001](https://user-images.githubusercontent.com/4016000/26906232-27274492-4bec-11e7-850f-31bcf4267608.jpg)
![screenshot0002](https://user-images.githubusercontent.com/4016000/26906236-2a7cbb90-4bec-11e7-9bcd-15780e03d3e4.jpg)

![screenshot0005](https://user-images.githubusercontent.com/4016000/26906243-39c5ad00-4bec-11e7-8d80-cd53c52c5019.jpg)
![screenshot0004](https://user-images.githubusercontent.com/4016000/26906245-3c3d9016-4bec-11e7-87bd-5372ea0a532a.jpg)

- USDX version: Master  /10 Apr 17/ 
- win 10 with enough PC power

**Comment by basisbit** [comment]
most parts of UltraStar Deluxe so far are not multi threaded. Thus, the application waits for the next webcam frame, shows it and only after that starts calculating and drawing the next frame.

Honestly, I doubt anyone will do the work to make this multi threaded, as the used development language objective pascal is not really "modern" in various ways. If anyone wants to implement this, please feel free to create a pull request so these changes can be merged into this repository.

**Comment by turric4n** [comment]
Object Pascal is a modern language and has every new language features has (dot). The webcam capture need to be refactored but not very hard to do it using Threading.

**Comment by basisbit** [comment]
just for documentation 
![image](https://user-images.githubusercontent.com/10870269/33887238-70be3016-df49-11e7-979d-7331422c7092.png)

**Comment by Thaddy** [comment]
Indeed. Object Pascal has the exact same threading functionality and synchronization primitives as e.g. C++.
There are also open source high level true multi-threading libraries available.
In the near future also high level language extensions like ParallelFor etc. The latter is already available in some threading libraries for FPC/Lazarus Object Pascal and in Delphi, which is also Object Pascal. For an experienced Object Pascal programmer it should be no problem to implement threaded frame rendering.
